File: src/main/java/com/github/tanokun/tanorpg/event/tanorpg/TanoRpgSellEvent.java
package com.github.tanokun.tanorpg.event.tanorpg;

import com.github.tanokun.tanorpg.player.Member;
import org.bukkit.entity.Player;
import org.bukkit.event.HandlerList;
import org.bukkit.event.player.PlayerEvent;
import org.jetbrains.annotations.NotNull;

public class TanoRpgSellEvent extends PlayerEvent {
    private static final HandlerList handlers = new HandlerList();

    private final Member member;

    public TanoRpgSellEvent(Player who, Member member) {
        super(who);
        this.member = member;
    }

    public static HandlerList getHandlerList() {
        return handlers;
    }

    @Override
    public @NotNull HandlerList getHandlers() {
        return handlers;
    }

    public Member getMember() {
        return member;
    }
}

File: src/main/java/com/github/tanokun/tanorpg/event/tanorpg/TanoRpgSellEvent.kt
package com.github.tanokun.tanorpg.event.tanorpg

import com.github.tanokun.tanorpg.player.Member
import org.bukkit.entity.Player
import org.bukkit.event.HandlerList
import org.bukkit.event.player.PlayerEvent

class TanoRpgSellEvent(who: Player?, val member: Member) : PlayerEvent(who!!) {

    override fun getHandlers(): HandlerList {
        return handlerList
    }

    companion object {
        private val handlerList = HandlerList()

        @JvmStatic
        fun getHandlerList(): HandlerList {
            return handlerList
        }
    }
}